debug_metadata/indexmap.natvis

```diff
@@ -0,0 +1,29 @@
+<AutoVisualizer xmlns="http://schemas.microsoft.com/vstudio/debugger/natvis/2010">
+  <Type Name="indexmap::set::IndexSet&lt;*,*&gt;">
```

Comment:
This is more a comment toward RFC 3191, I suppose, but I'm surprised to see unmangled path matching here, especially since the `indexmap` root could refer to multiple instances in the same binary, like 1.x and 2.x. I suppose to be more precise, you would need one of the listed alternative or future possibilities that puts the attribute directly on the type declaration.

Reply:
The type name for a Natvis definition matches against the debuginfo names embedded into the PDB. This does **not** contain any of the crate disambiguators you mention above in terms of version numbers. 

In the case where a binary contains multiple versions of a given crate, if both crates define the same type, the Natvis visualizer will match against both types.

Natvis does have support for `Conditions` which means certain fields or definitions can be gated upon whether or not a condition is met. This can help alleviate the issue of multiple versions of the same type.